ddcommon-ffi/src/metrics.rs

```diff
@@ -0,0 +1,76 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022-Present Datadog, Inc.
+
+use crate::{slice::AsBytes, slice::CharSlice};
+use ddcommon::metric::Metric;
+
+#[must_use]
+#[no_mangle]
+pub extern "C" fn ddog_Vec_Metric_new() -> crate::Vec<Metric> {
+    crate::Vec::default()
+}
+
+#[no_mangle]
+pub extern "C" fn ddog_Vec_Metric_drop(_: crate::Vec<Metric>) {}
+
+#[repr(C)]
+pub enum PushMetricResult {
+    Ok,
+    Err(crate::Vec<u8>),
+}
+
+#[no_mangle]
+pub extern "C" fn ddog_Vec_Metric_PushResult_drop(_: PushMetricResult) {}
+
+/// Creates a new Tag from the provided `key` and `value` by doing a utf8
+/// lossy conversion, and pushes into the `vec`. The strings `key` and `value`
+/// are cloned to avoid FFI lifetime issues.
+///
+/// # Safety
+/// The `vec` must be a valid reference.
+/// The CharSlices `key` and `value` must point to at least many bytes as their
+/// `.len` properties claim.
+#[no_mangle]
+pub unsafe extern "C" fn ddog_Vec_Metric_push(
+    vec: &mut crate::Vec<Metric>,
+    key: CharSlice,
+    value: f32,
```

Comment:
I don't know the answer: should this be `f64`? In PHP, tracer metrics are doubles, but that may because the language doesn't offer anything finer, I don't know.

Reply:
you are right. Will fix